pkgs/by-name/fl/flashprog/package.nix

```diff
@@ -24,26 +26,38 @@ stdenv.mkDerivation rec {
     installShellFiles
     pkg-config
   ];
-
   buildInputs = [
     libftdi1
-    libjaylink
     libusb1
   ] ++ lib.optionals (!stdenv.isDarwin) [
-    libgpiod_1
     pciutils
+  ] ++ lib.optionals with_jlink [
+    libjaylink
+  ] ++ lib.optionals (stdenv.isLinux && with_gpio) [
+    libgpiod
   ];
 
-  makeFlags = [ "PREFIX=$(out)" "libinstall" ]
-    ++ lib.optionals stdenv.isDarwin [ "CONFIG_ENABLE_LIBPCI_PROGRAMMERS=no" ]
-    ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [ "CONFIG_INTERNAL_X86=no" "CONFIG_INTERNAL_DMI=no" "CONFIG_RAYER_SPI=0" ];
+  makeFlags = [
+    "PREFIX=$(out)"
+    "libinstall"
+  ] ++ lib.optionals (!with_jlink) [
+    "CONFIG_JLINK_SPI=no"
+  ] ++ lib.optionals (!(stdenv.isLinux && with_gpio)) [
+    "CONFIG_LINUX_GPIO_SPI=no"
+  ] ++ lib.optionals stdenv.isDarwin [
+    "CONFIG_ENABLE_LIBPCI_PROGRAMMERS=no"
+  ] ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [
+    "CONFIG_INTERNAL_X86=no"
+    "CONFIG_INTERNAL_DMI=no"
+    "CONFIG_RAYER_SPI=no"
+  ];
 
-  meta = with lib; {
+  meta = {
     homepage = "https://flashprog.org";
     description = "Utility for reading, writing, erasing and verifying flash ROM chips";
-    license = with licenses; [ gpl2 gpl2Plus ];
-    maintainers = with maintainers; [ felixsinger ];
-    platforms = platforms.all;
+    license = with lib.licenses; [ gpl2 gpl2Plus ];
+    maintainers = with lib.maintainers; [ felixsinger ];
+    platforms = lib.platforms.all;
```

Comment:
It feels like people are going back and forth with the with statement. If the change is just because of personal preference, then please change it back.

Reply:
@AndersonTorres said in their review that nesting with statements is bad and I should change it to that. But I can also change it back if preferred.